Modules/xASL_module_ASL.m

```diff
@@ -429,20 +429,16 @@
     end
     
 	% allow 4D quantification as well, storing CBF4D. This is currently not implemented for multi-PLD/multi-TE (BASIL/FABBER)
-	if isfield(x.modules.asl, 'SaveCBF4D') && x.modules.asl.SaveCBF4D
-        if x.Q.nUniqueInitial_PLD>1 || x.modules.asl.bQuantifyMultiTE
-            warning('Saving CBF4D was requested but not implemented yet for multi-PLD or multi-TE, skipping');
+	if x.modules.asl.SaveCBF4D && x.Q.nUniqueInitial_PLD == 1 && x.modules.asl.bQuantifyMultiTE == false
```

Comment:
If we do a single-PLD of one of the multi-PLDs, you can do a CBF4D on this.

But I agree with you. I would just have the checks, warnings, etc above, so we can remove these redundant checks here and simplify the code. Otherwise, you can have really a lot of if statements throughout, because of course there many more reasons to skip a part of the code.

Reply:
Yes, true, a single-PLD can be selected, but that's probably not what we want to avoid confusion.

OK - no prob. The warning is redundant, so no point in keeping it. Done in 696ad616

I'm leaving this FYI. YOu can resolve without checking the code.